gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/pod/tasks/CreatePodYamlTask.java

```diff
@@ -0,0 +1,121 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist.pod.tasks;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.gradle.dist.pod.PodServiceDefinition;
+import com.palantir.gradle.dist.pod.PodVolumeDefinition;
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+public class CreatePodYamlTask extends DefaultTask {
+    private static final String VOLUME_NAME_REGEX = "^(?:[a-z0-9]+?-)*[a-z0-9]+$";
+    private static final String SERVICE_VALIDATION_FAIL_FORMAT = "Pod validation failed for service %s: %s";
+    private static final String VOLUME_VALIDATION_FAIL_FORMAT = "Pod validation failed for volume %s: %s";
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .setPropertyNamingStrategy(PropertyNamingStrategy.KEBAB_CASE)
+            .enable(SerializationFeature.INDENT_OUTPUT);
+
+    private MapProperty<String, PodServiceDefinition> serviceDefinitions =
+            getProject().getObjects().mapProperty(String.class, PodServiceDefinition.class);
+    private MapProperty<String, PodVolumeDefinition> volumeDefinitions =
+            getProject().getObjects().mapProperty(String.class, PodVolumeDefinition.class);
+    private RegularFileProperty podYamlFile = getProject().getObjects().fileProperty();
+
+    public CreatePodYamlTask() {
+        podYamlFile.set(new File(getProject().getBuildDir(), "deployment/pod.yml"));
+    }
+
+    @Input
+    public final MapProperty<String, PodServiceDefinition> getServiceDefinitions() {
+        return serviceDefinitions;
+    }
+
+    @Input
+    public final MapProperty<String, PodVolumeDefinition> getVolumeDefinitions() {
+        return volumeDefinitions;
+    }
+
+    @OutputFile
+    public final RegularFileProperty getPodYamlFile() {
+        return podYamlFile;
+    }
+
+    @TaskAction
+    final void createPodYaml() throws IOException {
+        validatePodYaml();
+        OBJECT_MAPPER.writeValue(getPodYamlFile().getAsFile().get(), ImmutableMap.of(
+                "services",
+                OBJECT_MAPPER.convertValue(this.serviceDefinitions.get(), new TypeReference<Map<String, Object>>() {}),
+                "volumes",
+                OBJECT_MAPPER.convertValue(this.volumeDefinitions.get(), new TypeReference<Map<String, Object>>() {})));
```

Comment:
Since the types are heterogeneous, It was between doing a convert values or creating an immutable for this single usage. Happy to switch over to immutable if you think its cleaner